File: SigreAppMobile/Hermes/app/src/main/java/pe/com/hermes/appmobile/ui/common/SimpleListAdapter.java
package pe.com.hermes.appmobile.ui.common;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;
import java.util.ArrayList;
import java.util.Collections;
import java.util.List;
import pe.com.hermes.appmobile.databinding.ItemSimpleTextoBinding;

/**
 * Adapter de listados Hermes (branding): filas pegadas, sin separación vertical entre registros.
 */
public class SimpleListAdapter extends RecyclerView.Adapter<SimpleListAdapter.VH> {

    public interface OnItemClick {
        void onClick(SimpleItem item);
    }

    private List<SimpleItem> items;
    private final OnItemClick onClick;

    public SimpleListAdapter(List<SimpleItem> items, OnItemClick onClick) {
        this.items = items != null ? items : Collections.emptyList();
        this.onClick = onClick;
    }

    public SimpleListAdapter(OnItemClick onClick) {
        this(new ArrayList<>(), onClick);
    }

    public void actualizar(List<SimpleItem> nuevos) {
        items = nuevos != null ? nuevos : Collections.emptyList();
        notifyDataSetChanged();
    }

    @Override
    public void onAttachedToRecyclerView(@NonNull RecyclerView recyclerView) {
        super.onAttachedToRecyclerView(recyclerView);
        HermesListUi.aplicarListaContinua(recyclerView);
    }

    @NonNull
    @Override
    public VH onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        ItemSimpleTextoBinding binding = ItemSimpleTextoBinding.inflate(
                LayoutInflater.from(parent.getContext()), parent, false);
        HermesListUi.anularMargenesVerticales(binding.getRoot().getLayoutParams());
        return new VH(binding);
    }

    @Override
    public void onBindViewHolder(@NonNull VH holder, int position) {
        HermesListUi.anularMargenesVerticales(holder.itemView.getLayoutParams());
        holder.bind(items.get(position), onClick);
    }

    @Override
    public int getItemCount() {
        return items.size();
    }

    static final class VH extends RecyclerView.ViewHolder {
        private final ItemSimpleTextoBinding binding;

        VH(ItemSimpleTextoBinding binding) {
            super(binding.getRoot());
            this.binding = binding;
        }

        void bind(SimpleItem item, OnItemClick onClick) {
            binding.tvTitulo.setText(item.titulo);
            binding.tvSubtitulo.setText(item.subtitulo != null ? item.subtitulo : "");
            binding.tvSubtitulo.setVisibility(
                    item.subtitulo == null || item.subtitulo.trim().isEmpty() ? View.GONE : View.VISIBLE
            );
            binding.getRoot().setOnClickListener(v -> onClick.onClick(item));
        }
    }
}

File: SigreAppMobile/SigreLogistica/app/src/main/java/pe/com/sigre/logistica/ui/common/SimpleListAdapter.kt
package pe.com.sigre.logistica.ui.common

import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import pe.com.sigre.logistica.databinding.ItemSimpleTextoBinding

/** Adapter genérico reutilizable para listas de selección (empresas, sucursales, opciones de menú). */
class SimpleListAdapter(
    private var items: List<SimpleItem> = emptyList(),
    private val onClick: (SimpleItem) -> Unit,
) : RecyclerView.Adapter<SimpleListAdapter.VH>() {

    fun actualizar(nuevos: List<SimpleItem>) {
        items = nuevos
        notifyDataSetChanged()
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): VH {
        val binding = ItemSimpleTextoBinding.inflate(LayoutInflater.from(parent.context), parent, false)
        return VH(binding)
    }

    override fun onBindViewHolder(holder: VH, position: Int) {
        holder.bind(items[position], onClick)
    }

    override fun getItemCount(): Int = items.size

    class VH(private val binding: ItemSimpleTextoBinding) : RecyclerView.ViewHolder(binding.root) {
        fun bind(item: SimpleItem, onClick: (SimpleItem) -> Unit) {
            binding.tvTitulo.text = item.titulo
            binding.tvSubtitulo.text = item.subtitulo.orEmpty()
            binding.tvSubtitulo.visibility = if (item.subtitulo.isNullOrBlank()) android.view.View.GONE else android.view.View.VISIBLE
            binding.root.setOnClickListener { onClick(item) }
        }
    }
}
